EarlyBird/CoursesTableViewController.swift

```diff
@@ -24,4 +28,57 @@ class CoursesTableViewController: UITableViewController {
         cell!.textLabel!.text = courseNames[indexPath.row]
         return cell!
     }
+    
+    func makeGetRequest() {
+        // Set up the URL request for one specific course to before tring all courses
+        let courseEndpoint: String = "https://early-bird-courses.herokuapp.com/api/v1/courses/1"
+        
+        guard let url = URL(string: courseEndpoint) else {
+            print("Error: cannot create URL")
+            return
+        }
+        
+        let urlRequest = URLRequest(url: url)
+        // set up the session
+        let config = URLSessionConfiguration.default
+        let session = URLSession(configuration: config)
+        
+        // make the GET request
+        let task = session.dataTask(with: urlRequest) { (data, response, error) in
+            // check for any errors
+            guard error == nil else {
+                print("error calling GET on /courses/1")
+                print(error)
+                return
+            }
+            // make sure data was returned
+            print("THIS IS THE DATA!!!\(data)")
+            
+            guard let responseData = data else {
+                print("Error: did not receive data")
+                return
+            }
+            // parse the result as JSON
+            do {
+                guard let course = try JSONSerialization.jsonObject(with: responseData, options: []) as? [String: AnyObject] else {
+                    print("error trying to convert data to JSON")
+                    return
+                }
+                // print the entire course dictionary
+                print("The course is: \(course)")
+                
+                // assign course["name"] to constant courseName
+                guard let courseName = course["name"] as? String else {
+                    print("Could not get course name from JSON")
+                    return
+                }
+                // prove that I can access the course name by printing courseName constant
+                print("The course is the course name: \(courseName)!!!")
+            } catch  {
```

Comment:
Thanks @dtinianow Swift makes it clear that errors must be handled properly, so in a way Swift forced me into it.  [The Docs](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/ErrorHandling.html) provide a good example under the "Handling Errors" section if you are interested.